include/powsybl/iidm/Network.hpp

```diff
@@ -161,6 +161,8 @@ class Network : public Container, public VariantManagerHolder {
 
     ConnectedComponentsManager& getConnectedComponentsManager();
 
+    std::set<Country> getCountries() const;
```

Comment:
I wonder if we should use a range here

Reply:
Agree, write such a code is better:
```
stdcxx::const_range<Country> Network::getCountries() const {
    const auto& filter = [](const Substation& substation) {
        return static_cast<bool>(substation.getCountry());
    };

    const auto& mapper = [](const Substation& substation) {
        return *substation.getCountry();
    };

    return getSubstations() | boost::adaptors::filtered(filter) | boost::adaptors::transformed(mapper);
}
```
But the problem is that we also have to ensure unicity, and 
```
    return boost::range::unique(getSubstations() | boost::adaptors::filtered(filter) | boost::adaptors::transformed(mapper));
```
does not compile because of a strange non assignable error (probably because it's a range of references):
```
[  0%] Building CXX object src/CMakeFiles/iidm.dir/iidm/Network.cpp.o
/usr/include/boost/optional/optional.hpp:697:58: error: object of type '(lambda at /mnt/e/Projet_SRO/rte/powsybl-iidm4cpp/src/iidm/Network.cpp:211:26)' cannot be assigned because its copy assignment operator is implicitly deleted [clang-diagnostic-error]
    void assign_value ( argument_type val ) { get_impl() = val; }
```

Both statements:
```
    return getSubstations() | boost::adaptors::filtered(filter) | boost::adaptors::transformed(mapper) | boost::adaptors::uniqued;
    return boost::adaptors::unique(getSubstations() | boost::adaptors::filtered(filter) | boost::adaptors::transformed(mapper));
```
compile but does nothing 😢